internal/pkg/services/argus/utils/utils.go

```diff
@@ -56,6 +57,143 @@ func LoadPlanId(planName string, resp *argus.PlansResponse) (*string, error) {
+		configs := make([]argus.CreateScrapeConfigPayloadMetricsRelabelConfigsInner, 0)
+		for _, config := range *data.MetricsRelabelConfigs {
+			configs = append(configs, mapMetricsRelabelConfig(config))
+		}
+		metricsRelabelConfigs = &configs
+	}
+
+	var params *map[string]interface{}
+
+	if data.Params != nil {
+		paramsMap, err := convertMapAnyToInterface(*data.Params)
+		if err != nil {
+			return nil, fmt.Errorf("convert params: %w", err)
+		}
+		params = &paramsMap
+	}
+
+	return &argus.UpdateScrapeConfigPayload{
+		BasicAuth:             basicAuth,
+		BearerToken:           data.BearerToken,
+		HonorLabels:           data.HonorLabels,
+		HonorTimeStamps:       data.HonorTimeStamps,
+		MetricsPath:           data.MetricsPath,
+		MetricsRelabelConfigs: metricsRelabelConfigs,
+		Params:                params,
+		SampleLimit:           convertIntToFloat64(data.SampleLimit),
+		Scheme:                data.Scheme,
+		ScrapeInterval:        data.ScrapeInterval,
+		ScrapeTimeout:         data.ScrapeTimeout,
+		StaticConfigs:         staticConfigs,
+		TlsConfig:             tlsConfig,
+	}, nil
+}
+
+func convertIntToFloat64(i *int64) *float64 {
+	if i == nil {
+		return nil
+	}
+	f := float64(*i)
+	return &f
+}
+
+func mapMetricsRelabelConfig(metricsRelabelConfig argus.MetricsRelabelConfig) argus.CreateScrapeConfigPayloadMetricsRelabelConfigsInner {
+	return argus.CreateScrapeConfigPayloadMetricsRelabelConfigsInner{
+		Action:       metricsRelabelConfig.Action,
+		Modulus:      convertIntToFloat64(metricsRelabelConfig.Modulus),
+		Regex:        metricsRelabelConfig.Regex,
+		Replacement:  metricsRelabelConfig.Replacement,
+		Separator:    metricsRelabelConfig.Separator,
+		SourceLabels: metricsRelabelConfig.SourceLabels,
+		TargetLabel:  metricsRelabelConfig.TargetLabel,
+	}
+}
+
+func mapStaticConfig(staticConfig argus.StaticConfigs) (argus.UpdateScrapeConfigPayloadStaticConfigsInner, error) {
+	var labels map[string]interface{}
+	var err error
+	if staticConfig.Labels != nil {
+		labels, err = convertMapAnyToInterface(*staticConfig.Labels)
+
+		if err != nil {
+			return argus.UpdateScrapeConfigPayloadStaticConfigsInner{}, fmt.Errorf("convert labels: %w", err)
+		}
+	}
+
+	return argus.UpdateScrapeConfigPayloadStaticConfigsInner{
+		Labels:  &labels,
+		Targets: staticConfig.Targets,
+	}, nil
+}
+
+func mapBasicAuth(basicAuth *argus.BasicAuth) *argus.CreateScrapeConfigPayloadBasicAuth {
+	if basicAuth == nil {
+		return nil
+	}
+
+	return &argus.CreateScrapeConfigPayloadBasicAuth{
+		Password: basicAuth.Password,
+		Username: basicAuth.Username,
+	}
+}
+
+func mapTlsConfig(tlsConfig *argus.TLSConfig) *argus.CreateScrapeConfigPayloadHttpSdConfigsInnerOauth2TlsConfig {
+	if tlsConfig == nil {
+		return nil
+	}
+
+	return &argus.CreateScrapeConfigPayloadHttpSdConfigsInnerOauth2TlsConfig{
+		InsecureSkipVerify: tlsConfig.InsecureSkipVerify,
+	}
+}
+
+func convertMapAnyToInterface(m interface{}) (map[string]interface{}, error) {
```

Comment:
wouldn't it make more sense to have a function to map specifically `params` and `staticConfig.Labels` instead of this generic one? 
We should avoid receiving interface as an argument

Reply:
The only reason I did it this way is that the conversion code is the same, but I agree that the code is not the clearest. Updated it.